lib/matplotlib/gridspec.py

```diff
@@ -309,23 +309,11 @@ def subplots(self, *, sharex=False, sharey=False, squeeze=True,
 
         # turn off redundant tick labeling
         if sharex in ["col", "all"]:
-            for ax in axarr[:-1, :].flat:  # Remove bottom labels/offsettexts.
-                ax.xaxis.set_tick_params(which="both", labelbottom=False)
```

Comment:
This would change visibility, I think? But `label_top` sets an empty label instead. Is that completely equivalent? Could it have some effect on tight layout or similar?

Reply:
Are you talking about the tick labels or the axis labels?  If it's the axis labels there's a single Text object per axis (`axis.label`, see e.g. `Axis.get_label`) which is always visible, just possibly an empty string (check e.g. `gca().xaxis.get_label().get_visible()`).

Reply:
I meant the Axis labels, but yes, I had confused the tick's `labeltop` setting with them.